applications/simulate_prod.py

```diff
@@ -0,0 +1,332 @@
+from simtools.configuration import configurator
+from simtools.configuration.commandline_parser import CommandLineParser
+from simtools.simulator import Simulator
+
+
+def _parse(description=None):
+    """
+    Parse the command line configuration.
+
+    Parameters
+    ----------
+    description: str
+        Description of the application.
+
+    Returns
+    -------
+    CommandLineParser
+        Command line parser object.
+
+    """
+
+    config = configurator.Configurator(description=description)
+    config.parser.add_argument(
+        "--production_config",
+        help=(
+            "Simulation configuration file "
+            "(contains the default setup which can be overwritten by the command line options)"
+        ),
+        type=str,
+        required=True,
+    )
+    config.parser.add_argument(
+        "--primary",
+        help="Primary particle to simulate.",
+        type=str.lower,
+        required=True,
+        choices=[
+            "gamma",
+            "gamma_diffuse",
+            "electron",
+            "proton",
+            "muon",
+            "helium",
+            "nitrogen",
+            "silicon",
+            "iron",
+        ],
+    )
+    config.parser.add_argument(
+        "--from_azimuth_direction",
+        help="Direction from which the primary reaches the atmosphere",
+        type=str.lower,
+        required=True,
+        choices=[
+            "north",
+            "south",
+            "east",
+            "west",
+        ],
+    )
+    config.parser.add_argument(
+        "--zenith_angle",
+        help="Zenith angle in degrees",
+        type=CommandLineParser.zenith_angle,
+        required=True,
+    )
+    config.parser.add_argument(
+        "--nshow",
+        help="Number of showers to simulate",
+        type=int,
+        required=False,
+    )
+    config.parser.add_argument(
+        "--start_run",
+        help=(
+            "Start run number such that the actual run number will be 'start_run' + 'run'. "
+            "This is useful in case a new transform is submitted for the same production. "
+            "It allows the transformation system to keep using sequential run numbers without "
+            "repetition."
+        ),
+        type=int,
+        required=True,
+    )
+    config.parser.add_argument(
+        "--run",
+        help="Run number (actual run number will be 'start_run' + 'run')",
+        type=int,
+        required=True,
+    )
+    config.parser.add_argument(
+        "--data_directory",
+        help="The directory where to save the corsika-data and simtel-data output directories.",
+        type=str.lower,
+        required=False,
+        default="./",
+    )
+    return config.initialize(db_config=True, telescope_model=True)
+
+
+def _proccess_simulation_config_file(config_file, primary_config, logger):
+    """
+    Read the simulation configuration file with all the details
+    on shower and array simulations.
+
+    Attributes
```

Comment:
these are not attributes, but parameters, no? We do not list attributes.

The _logger is missing

Reply:
You are right, fixed.